### PR TITLE
Feature/honzon bridge

### LINKED ISSUE #1949: module-honzon-bridge
- module-honzon-bridge
  - Types
  - Config
    - type Currency: MultiCurrency
    - constant StablecoinCurrencyId: Get\<CurrencyId>
    - constant BridgedStableCurrencyId: Get\<CurrencyId>
    - constant PalletId: Get\<PalletId>
  - Calls
    - fn to_bridged(origin, amount: Balance)
      - ensure signed
      - transfer `amount` of `StablecoinCurrencyId` to `PalletId` account
      - transfer `amount` of `BridgedStableCurrencyId` from `PalletId` account to `origin`
    - fn from_bridged(origin)
      - ensure signed
      - transfer `amount of ` `BridgedStableCurrencyId` to `PalletId` account
      - transfer `amount` of `StablecoinCurrencyId` from `PalletId` account to `origin`

And update transaction payment to special case the wormholeAusd ERC20 token use this pallet to swap it to kUSD for tx fee paying. This will allow people bringing aUSD into Karura can use the bridged token for tx fee to call `from_bridged` to redeem their token.

### PR DESCRIPTION
**Pull request by syan095**
Added the new Honzon bridge module, where user can exchange Bridged wormhole AUSD with native KUSD on the karura runtime.

Closes #1949

### CODE REVIEW

File: modules/honzon-bridge/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		#[pallet::weight(< T as Config >::WeightInfo::to_bridged())]
+		#[transactional]
+		pub fn to_bridged(origin: OriginFor<T>, #[pallet::compact] amount: Balance) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// transfer amount of StablecoinCurrencyId to PalletId account
+			T::Currency::transfer(T::StablecoinCurrencyId::get(), &who, &Self::account_id(), amount)?;
+
+			// transfer amount of BridgedStableCoinCurrencyId from PalletId account to origin
+			T::Currency::transfer(T::BridgedStableCoinCurrencyId::get(), &Self::account_id(), &who, amount)?;
```
Comment (zjb0807) [inline modules/honzon-bridge/src/lib.rs:102]: should use evm-bridge 
https://github.com/AcalaNetwork/Acala/blob/bf3d860f388117b8bb74502795be38704fdacfcd/modules/evm-bridge/src/lib.rs#L172

File: runtime/karura/src/lib.rs [hunk 2]
```diff
@@ -1551,6 +1553,20 @@ impl module_idle_scheduler::Config for Runtime {
 	type MinimumWeightRemainInBlock = MinimumWeightRemainInBlock;
 }
 
+parameter_types! {
+	pub WormholeAUSDCurrencyId: CurrencyId = CurrencyId::Erc20(EvmAddress::from(hex_literal::hex!["0000000000000000000100000000000000000001"]));
```
Comment (zjb0807) [inline runtime/karura/src/lib.rs:1557]: Create an issue and update here after creating waUSD

Comment (zjb0807) [reply runtime/karura/src/lib.rs:1557]: And disable it here:
https://github.com/AcalaNetwork/Acala/blob/bf3d860f388117b8bb74502795be38704fdacfcd/runtime/karura/src/lib.rs#L213-L220

File: runtime/karura/src/lib.rs [hunk 3]
```diff
@@ -2218,7 +2236,7 @@ mod tests {
 	#[test]
 	fn check_call_size() {
 		assert!(
-			core::mem::size_of::<Call>() <= 260,
+			core::mem::size_of::<Call>() <= 300,
```
Comment (xlc) [inline runtime/karura/src/lib.rs:2239]: why increased?

Comment (syan095) [reply runtime/karura/src/lib.rs]: This is commited by mistake. I'll revert this

File: runtime/karura/src/lib.rs (hunk 2 above)
Comment (syan095) [reply runtime/karura/src/lib.rs:1557]: Created an issue here: 
https://github.com/AcalaNetwork/Acala/issues/1967

Comment (syan095) [reply runtime/karura/src/lib.rs:1558]: Issue created and HonzonBridge call disabled

File: modules/honzon-bridge/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+		///
+		/// Parameters:
+		/// - `amount`: The amount of stable coin to exchange.
+		#[pallet::weight(< T as Config >::WeightInfo::to_bridged())]
+		#[transactional]
+		pub fn to_bridged(origin: OriginFor<T>, #[pallet::compact] amount: Balance) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// transfer amount of StablecoinCurrencyId to PalletId account
+			T::Currency::transfer(T::StablecoinCurrencyId::get(), &who, &Self::account_id(), amount)?;
```
Comment (xlc) [inline modules/honzon-bridge/src/lib.rs:99]: Better to cache result from `Self::account_id()`

https://github.com/paritytech/substrate/blob/345e3b1fcbbb0c479a031c45859d7154e7c4b5f7/frame/treasury/src/lib.rs#L404-L405

**Review by xlc — APPROVED** [review]

**Pull request merged**